docs/examples/plot_interval_transposition_error.py

```diff
@@ -0,0 +1,157 @@
+"""
+Modeling with interval averages
+===============================
+
+Transposing interval-averaged irradiance data
+"""
+
+# %%
+# This example shows how not accounting for the difference between
+# instantaneous and interval-averaged time series data can introduce
+# error in the modeling process. An instantaneous time series
+# represents discrete measurements taken at each timestamp, while
+# an interval-averaged time series represents the average value across
+# each data interval.  For example, the value of an interval-averaged
+# hourly time series at 11:00 represents the average value between
+# 11:00 and 12:00 (if the series is left-labeled), the average value
+# between 10:00 and 11:00 (if the series is right-labeled), or the
+# average value between 10:30 and 11:30 (if the series is
+# center-labeled).  Interval-averaged time series are common in
+# field data, where the datalogger averages high-frequency measurements
+# into low-frequency averages for archiving purposes.
+#
+# It is important to account for this difference When using
+# interval-averaged weather data for modeling.  This example
+# focuses on calculating solar position appropriately for
+# irradiance transposition, but this concept is relevant for
+# other steps in the modeling process as well.
+#
+# This example calculates a POA irradiance timeseries at 1-second
+# resolution as a "ground truth" value.  Then it performs the
+# transposition again at lower resolution using interval-averaged
+# irradiance components, once using a half-interval shift and
+# once just using the unmodified timestamps. The difference
+# affects the solar position calculation: for example, assuming
+# we have average irradiance for the interval 11:00 to 12:00,
+# and it came from a left-labeled time series, naively using
+# the unmodified timestamp will calculate solar position for 11:00,
+# meaning the calculated solar position is used to represent
+# times as far as an hour away.  A better option would be to
+# calculate the solar position at 11:30 to reduce the maximum
+# timing error to only half an hour.
+
+import pvlib
+import pandas as pd
+import matplotlib.pyplot as plt
+
+# %%
+# First, we'll define a helper function that we can re-use several
+# times in the following code:
+
+
+def transpose(irradiance, timeshift):
+    """
+    Transpose irradiance components to plane-of-array, incorporating
+    a timeshift in the solar position calculation.
+
+    Inputs:
```

Comment:
mild preference for using the numpydoc format since that's what pvlib users should be used to reading.